pkg/manager/member/pd_upgrader.go

```diff
@@ -19,13 +19,19 @@ import (
 	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
 	"github.com/pingcap/tidb-operator/pkg/controller"
 	mngerutils "github.com/pingcap/tidb-operator/pkg/manager/utils"
+	"github.com/pingcap/tidb-operator/pkg/pdapi"
 
 	"github.com/pingcap/advanced-statefulset/client/apis/apps/v1/helper"
 	apps "k8s.io/api/apps/v1"
 	"k8s.io/klog/v2"
 	podutil "k8s.io/kubernetes/pkg/api/v1/pod"
 )
 
+const (
+	// set this PD clustre annotation to true to fail cluster upgrade if PD loose the quorum during one pod restart
+	annoKeyPDPeersCheck = "tidb.pingcap.com/pd-check-quorum-before-upgrade"
```

Comment:
Minior comments. Before the change seems we have already performed some quorum checks but using CR status instead of PD client, maybe we can be more specific in the annotation, telling people that when the annotation is set we're using PD client?